### ISSUE TITLE
Form value changed after upgrading from `2.1.4` to `2.1.5`.

### ISSUE THREAD

**Issue by yujonglee** [ash-project/ash_phoenix#253]
**Describe the bug**

When I upgrade from `2.1.4` to `2.1.5`, I see some behavior change with form.
(same in `2.1.6)

https://github.com/fastrepl/canary/blob/e637cde/core/lib/canary_web/live/source_live/detail.ex#L65

In `2.1.4`, `fc[:start_urls].value` this contains correct existing value(string array).

<details>

<summary>In `2.1.5`, it is nil.</summary>

```elixir
%Phoenix.HTML.FormField{
  id: "form_config_0_start_urls",
  name: "form[config][start_urls]",
  errors: [],
  field: :start_urls,
  form: %Phoenix.HTML.Form{
    source: #AshPhoenix.Form<
      resource: AshPhoenix.Form.WrappedValue,
      action: :update,
      type: :update,
      params: %{},
      source: #Ash.Changeset<
        domain: Canary.Sources,
        action_type: :update,
        action: :update,
        attributes: %{},
        relationships: %{},
        errors: [],
        data: #AshPhoenix.Form.WrappedValue<
          __meta__: #Ecto.Schema.Metadata<:built, "">,
          value: %Ash.Union{
            value: #Canary.Sources.Webpage.Config<
              __meta__: #Ecto.Schema.Metadata<:built, "">,
              start_urls: ["https://litellm2.ai", "https://litellm3.ai"],
              url_include_patterns: [],
              url_exclude_patterns: [],
              tag_definitions: [],
              js_render: false,
              aggregates: %{},
              calculations: %{}
            >,
            type: :webpage
          },
          aggregates: %{},
          calculations: %{}
        >,
        valid?: true
      >,
      name: "form[config]",
      data: #AshPhoenix.Form.WrappedValue<
        __meta__: #Ecto.Schema.Metadata<:built, "">,
        value: %Ash.Union{
          value: #Canary.Sources.Webpage.Config<
            __meta__: #Ecto.Schema.Metadata<:built, "">,
            start_urls: ["https://litellm2.ai", "https://litellm3.ai"],
            url_include_patterns: [],
            url_exclude_patterns: [],
            tag_definitions: [],
            js_render: false,
            aggregates: %{},
            calculations: %{}
          >,
          type: :webpage
        },
        aggregates: %{},
        calculations: %{}
      >,
      form_keys: [
        tag_definitions: [
          type: :list,
          resource: Canary.Sources.Webpage.TagDefinition,
          create_action: :create,
          update_action: :update,
          embed?: true,
          data: #Function<40.109777359/1 in AshPhoenix.Form.Auto.embedded/3>,
          forms: []
        ]
      ],
      forms: %{tag_definitions: []},
      domain: Canary.Sources,
      method: "put",
      submit_errors: nil,
      id: "form_config",
      transform_errors: nil,
      original_data: #AshPhoenix.Form.WrappedValue<
        __meta__: #Ecto.Schema.Metadata<:built, "">,
        value: %Ash.Union{
          value: #Canary.Sources.Webpage.Config<
            __meta__: #Ecto.Schema.Metadata<:built, "">,
            start_urls: ["https://litellm2.ai", "https://litellm3.ai"],
            url_include_patterns: [],
            url_exclude_patterns: [],
            tag_definitions: [],
            js_render: false,
            aggregates: %{},
            calculations: %{}
          >,
          type: :webpage
        },
        aggregates: %{},
        calculations: %{}
      >,
      transform_params: #Function<36.109777359/3 in AshPhoenix.Form.Auto.unions/3>,
      prepare_params: nil,
      prepare_source: nil,
      warn_on_unhandled_errors?: true,
      any_removed?: false,
      added?: false,
      changed?: false,
      touched_forms: MapSet.new([]),
      valid?: true,
      errors: false,
      submitted_once?: false,
      just_submitted?: false
    >,
    impl: Phoenix.HTML.FormData.AshPhoenix.Form,
    id: "form_config_0",
    name: "form[config]",
    data: #AshPhoenix.Form.WrappedValue<
      __meta__: #Ecto.Schema.Metadata<:built, "">,
      value: %Ash.Union{
        value: #Canary.Sources.Webpage.Config<
          __meta__: #Ecto.Schema.Metadata<:built, "">,
          start_urls: ["https://litellm2.ai", "https://litellm3.ai"],
          url_include_patterns: [],
          url_exclude_patterns: [],
          tag_definitions: [],
          js_render: false,
          aggregates: %{},
          calculations: %{}
        >,
        type: :webpage
      },
      aggregates: %{},
      calculations: %{}
    >,
    action: nil,
    hidden: [
      {"_persistent_id", "0"},
      {:_form_type, "update"}
    ],
    params: %{"_persistent_id" => "0"},
    errors: [],
    options: [method: "put"],
    index: 0
  },
  value: nil
}

```

</details>

Not sure what possibly cause some change.
https://github.com/ash-project/ash_phoenix/compare/v2.1.4...v2.1.5

**To Reproduce**

I don't have it yet.

**Expected behavior**

Should not have regression.

**Runtime**
 - Elixir version: 1.17.2 
 - Erlang version: 27.0.1
 - OS: Mac
 - Ash version: 3.4.33
 - any related extension versions

**Additional context**

**Comment by zachdaniel** [comment]
🤔 would you be able to provide a reproduction of this? Or provide instructions of how I can reproduce this in your project? I don't see how this would have affected other types of forms.

**Comment by zachdaniel** [comment]
Ideally a failing test in the `ash_phoenix` test suite.

**Comment by zachdaniel** [comment]
Okay, so this is a bit complicated. I may not have fixed every issue that arises from the `2.1.5` changes, but those changes were necessary for union forms. Try the changes in `main` and LMK.
